### ISSUE TITLE
Serilog isn't working on Debian

### ISSUE THREAD

**Issue by Techtonictools** [serilog/serilog#1443]
I'm unable to get Serilog to log information to a file or console on Linux. The log file is present on Linux, but it is empty. Serilog works fine on Windows. Same exact code. Same dotnet version. To deploy on Linux, I built using dotnet publish -c Debug -r linux-x64 then copied the binaries over to the the Linux machine and executed the binary both as a regular user and as root. It does not appear to be a permissions issue, but I'd like it to be my fault and quick to fix.

    
    class Program {
            public static async Task Main(string[] args) {
                NpgsqlConnection.GlobalTypeMapper.UseNetTopologySuite();
                Log.Logger = new LoggerConfiguration()
                    .MinimumLevel.Debug()
                    .MinimumLevel.Override("Microsoft", LogEventLevel.Information)
                    .Enrich.FromLogContext()
                    .WriteTo.Console()
                    .WriteTo.RollingFile(Path.Combine(AppContext.BaseDirectory, "Blah.txt"))
                    .CreateLogger();
                await CreateHostBuilder(args).Build().RunAsync();
            }
            public static IHostBuilder CreateHostBuilder(string[] args) {
                return Host.CreateDefaultBuilder(args)
                    .UseSerilog()
                    .ConfigureServices((hostContext, services) => {
                        services.AddOptions();
                        services.AddHostedService<BlahService>();
                    })
                    .ConfigureAppConfiguration((hostingContext, config) => {
                        config.AddEnvironmentVariables();
                    });
            }
        } 
    }

    public async Task StartAsync(CancellationToken cancellationToken) {
        try {
            Log.Information("Starting blah");
            var wxModels = await BlahModel.GetModelsAsync();
            // snipped
        }
        catch (Exception ex) {
            Log.Error(ex, string.Empty);
        }
    }

This is the expected output I get on Windows both from the console and in the log file:
   
    2020-06-04 14:30:24.117 -06:00 [Information] Starting blah
    2020-06-04 14:30:24.648 -06:00 [Error] 
    Npgsql.NpgsqlException (0x80004005): No password has been provided but the backend requires one (in MD5)
    at Npgsql.NpgsqlConnector.AuthenticateMD5(String username, Byte[] salt, Boolean async)
    at Npgsql.NpgsqlConnector.Authenticate(String username, NpgsqlTimeout timeout, Boolean async)
    at Npgsql.NpgsqlConnector.Open(NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
    at Npgsql.ConnectorPool.AllocateLong(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
    at Npgsql.NpgsqlConnection.<>c__DisplayClass32_0.<<Open>g__OpenLong|0>d.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
    at Blah.ActiveRecord.GetOpenedConnAsync() in D:\Src\Beta\blah\ActiveRecord.cs:line 179
    2020-06-04 14:30:24.791 -06:00 [Fatal] Critical: Failed to get weather models to harvest from database.
    2020-06-04 14:30:24.792 -06:00 [Information] Blah service started.
    2020-06-04 14:30:24.798 -06:00 [Information] Application started. Press Ctrl+C to shut down.
    2020-06-04 14:30:24.801 -06:00 [Information] Hosting environment: "Production"
    2020-06-04 14:30:24.803 -06:00 [Information] Content root path: "D:\Src\Beta\blah\bin\x64\Debug\netcoreapp3.1"
    2020-06-04 14:30:59.870 -06:00 [Information] Application is shutting down...
    2020-06-04 14:30:59.881 -06:00 [Information] Blah stopped.`

In debian, the serilog file exists but it is empty and 0 bytes in size. Launching VIA console:
    
    root@debian:/wx/blah# ./Blah
    info: Microsoft.Hosting.Lifetime[0]
        Application started. Press Ctrl+C to shut down.
    info: Microsoft.Hosting.Lifetime[0]
        Hosting environment: Production
    info: Microsoft.Hosting.Lifetime[0]
        Content root path: /wx/Blah
    ^Cinfo: Microsoft.Hosting.Lifetime[0]
        Application is shutting down...
    root@debian:/wx/Blah#
`

Version information:

    
    Serilog v2.9.0
    Serilog.AspNetCore v3.20
    Runtime Environment:
    OS Name:     debian
    OS Version:  10.4
    OS Platform: Linux
    RID:         debian.10-x64
    Base Path:   /usr/share/dotnet/sdk/3.1.300/
    Host (useful for support):
    Version: 3.1.4
    Commit:  0090613580 
    .NET Core SDKs installed:
    3.1.300 [/usr/share/dotnet/sdk]
    .NET Core runtimes installed:
    Microsoft.AspNetCore.App 3.1.4 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
    Microsoft.NETCore.App 3.1.4 [/usr/share/dotnet/shared/Microsoft.NETCore.App]`

**Comment by bartelink** [comment]
1. search up `SelfLog` re troubleshooting issues
2. `RollingFile` has been superseded by `File` (Serilog.Sinks.File)
3. Given the usage level of Serilog, you can trust that the relevant repo for the sink you are using will have an open issue if there is an actual bug

Aaaand, my wild guess - the path you are computing is not sufficiently privileged in your execution context 
And my proposal re how to analyze that - debug it as you would any other fine access issue on debian - neither Serilog nor .NET are going to be doing anything magical, and learning the techniques for debugging these things will be worth your time in the medium term

**Comment by Techtonictools** [comment]
Added 
    Serilog.Debugging.SelfLog.Enable(msg => Debug.WriteLine(msg)); 
to the program startup [per docs](https://github.com/serilog/serilog/wiki/Debugging-and-Diagnostics) and there was no change. The console output is the same as what is in the original post.

**Comment by bartelink** [comment]
Same w/ Console.WriteLine ?

**Comment by Techtonictools** [comment]
I changed from using .WriteTo.RollingFile to .WriteTo.File and the behavior is the same. The log file is present with the correct name.date format, but empty. There is no appsettings.* file present, I didn't need to remove the default logging entries.

**Comment by nblumhardt** [comment]
Strange! I'm running Serilog on Linux with .NET Core every day, so probably not a simple compat issue - the lack of any effect of `UseSerilog()` suggests there might be some kind of deployment issue, I know you've probably checked over these things @Techtonictools but it may need another pass - there's nothing in your code sample or scenario that would cause the issue, so unfortunately we're not likely to track it down, here.

**Comment by Techtonictools** [comment]
What deployment issues should I verify? I'm not sure what to look for. 
I've built several services for Linux on .net and the steps were just to copy the publish/* folder over and run it (minus some app settings/config file adjustments and creating a Linux service). Maybe it is a debian issue or a dotnet issue. I can install a known good build of Linux and .net core version on a new VM and try the program on it to try to narrow down the problem, I just need to know what OS & .net versions.

**Comment by Techtonictools** [comment]
@bartelink, I just saw your update after attempting to re-install Serilog based on nblumbhardts suggestion of a deployment issue. But I'm now unable to install Serilog at all. Feel free to close the bug.


    
    NU1202: Package Serilog.AspNetCore 3.2.0 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Serilog.AspNetCore 3.2.0 does not support any target frameworks.
    NU1202: Package Microsoft.AspNetCore.Hosting.Abstractions 2.0.0 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Microsoft.AspNetCore.Hosting.Abstractions 2.0.0 does not support any target frameworks.
    NU1202: Package Serilog.Extensions.Hosting 3.0.0 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Serilog.Extensions.Hosting 3.0.0 does not support any target frameworks.
    NU1202: Package Serilog.Formatting.Compact 1.0.0 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Serilog.Formatting.Compact 1.0.0 does not support any target frameworks.
    NU1202: Package Serilog.Settings.Configuration 3.1.0 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Serilog.Settings.Configuration 3.1.0 does not support any target frameworks.
    NU1202: Package Serilog.Sinks.Console 3.1.1 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Serilog.Sinks.Console 3.1.1 does not support any target frameworks.
    NU1202: Package Serilog.Sinks.Debug 1.0.1 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Serilog.Sinks.Debug 1.0.1 does not support any target frameworks.
    NU1202: Package Serilog.Sinks.File 4.0.0 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Serilog.Sinks.File 4.0.0 does not support any target frameworks.
    NU1202: Package Microsoft.AspNetCore.Hosting.Server.Abstractions 2.0.0 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Microsoft.AspNetCore.Hosting.Server.Abstractions 2.0.0 does not support any target frameworks.
    NU1202: Package Microsoft.AspNetCore.Http.Abstractions 2.0.0 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Microsoft.AspNetCore.Http.Abstractions 2.0.0 does not support any target frameworks.
    NU1202: Package Microsoft.AspNetCore.Http.Features 2.0.0 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Microsoft.AspNetCore.Http.Features 2.0.0 does not support any target frameworks.
    NU1202: Package System.Text.Encodings.Web 4.4.0 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package System.Text.Encodings.Web 4.4.0 does not support any target frameworks.
    Package restore failed. Rolling back package changes for 'blah'.

**Comment by nblumhardt** [comment]
Okay, thanks for the follow-up! Good luck with the tooling fix 👍

**Comment by Techtonictools** [comment]
@nblumhardt @bartelink the tooling issue re-installing Serilog (incompatibility) via nuget was solved. I re-deployed to Linux and Serilog is now logging both to file and the Console for me on debian. No code changes were made at all. So it must have been a nuget caching problem. 
 
When searching for these problems and Serilog I found the same (general) issues, but no resolutions. Both issues were resolved by clearing the nuget cache. In VS 2019 (v16.6.1), Tools->Nuget Package Manager Settings->Clear All Nuget Cache(s). I appreciate your work on Serilog.
